packages/emmett-fastify/src/index.ts

```diff
@@ -0,0 +1,101 @@
+import Compress from '@fastify/compress';
+import Etag from '@fastify/etag';
+import Form from '@fastify/formbody';
+import closeWithGrace from 'close-with-grace';
+import Fastify, {
+  type FastifyInstance,
+  type FastifyReply,
+  type FastifyRequest,
+} from 'fastify';
+
+const defaultPlugins = [
+  { plugin: Etag, options: {} },
+  { plugin: Compress, options: { global: false } },
+  { plugin: Form, options: {} },
+];
+
+const defaultPostMiddlewares = (app: FastifyInstance) => {
+  app.all('*', async (request: FastifyRequest, reply: FastifyReply) => {
+    return reply.status(404).send();
+  });
+};
+
+export interface ApplicationOptions {
+  serverOptions?: { logger: boolean };
+  registerRoutes?: (app: FastifyInstance) => void;
+  registerPreMiddlewares?: (app: FastifyInstance) => void;
```

Comment:
**SUGGESTION:** Unless Fastify expects to have middleware registered in a certain order (e.g. before adding hooks, then I'd prefer not to have that middleware and extends here. The' getApplication' intends to provide the safe defaults and the most common configuration, but allow users to use native way of configuring tooling. So I'd prefer to just pass the settings that Emmett will be responsible for, so let user do something like:


````ts
const application = getApplication({apis: [shoppingCartApi(eventStore, getUnitPrice, () => now)]});

// do additional registration here
fastify.use(require('something')())
````